### PR TITLE
add geospatial check table

### PR DESCRIPTION
**Pull request by Oysters1874**
Addresses the issue[ here](https://github.com/NYCPlanning/data-engineering-qaqc/issues/177). Thanks to Te's great work, we display two parts for the geospatial check, one is to check whether all properties are within NYC borough boundaries water included, the other is to check whether there is any inconsistency between geographies.

### CODE REVIEW

**Review by abrieff — COMMENTED** [review]
Would run the black linter on this think there are some missing formatting things but otherwise looks good

**Review by SashaWeinstein — COMMENTED** [review]
This is a good start! Have some comments I'll write out

**Comment by SashaWeinstein** [comment]
This is how I got `get_data` to run under the current organization of DO:
```python
    if branch == "dev":
        url = f"https://edm-publishing.nyc3.digitaloceanspaces.com/db-colp/{branch}/latest/output/qaqc"
    else:
        url = f"https://edm-publishing.nyc3.digitaloceanspaces.com/db-colp/{branch}/latest/output"

```
Obviously not ideal lol. We could limit the user to seeing dev for now, that might be the easiest way forward

**Comment by SashaWeinstein** [comment]
I have 3 comments on how the tables are laid out:
1. Can we put `uid` all the way to the right? So a user can scroll to it if they find the record they are looking for, or want to do more research. Having it on the left takes the place of more helpful columns
2. Can we recast version to be a human readable year-month?
3. Can we add an `expected_bbl_boro` and `expected_cd_boro` to the **Properties That Have Inconsistent Geographies** table? This might be a colp-side upgrade

File: src/colp/components/geospatial_check.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        self.display_not_in_nyc(all_records[0])
+        self.display_nonmatch(all_records[1])
+
+    def cast_version(self, v):
+        if len(v) == 6:
+            return str(v[:4]) + "-" + str(v[4:])
+
+    def display_not_in_nyc(self, records):
+        st.markdown("#### Properties That Are Not in NYC Borough Boundaries")
+        records = records["result"][0]["values"]
```
Comment (SashaWeinstein) [inline src/colp/components/geospatial_check.py:27]: I think we could add text between the header and the plot the way the **Properties That Have Inconsistent Geographies** report does

File: src/colp/components/geospatial_check.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            uid = df["uid"]
+            df = df.drop(columns="uid")
+            df = pd.concat([df, uid], axis=1)
+            df["v"] = df["v"].apply(lambda x: self.cast_version(x))
+            st.write(df)
+        else:
+            st.info("All properties are within NYC Borough Boundaries.")
+
+    def display_nonmatch(self, records):
+        st.markdown("#### Properties That Have Inconsistent Geographies")
```
Comment (SashaWeinstein) [inline src/colp/components/geospatial_check.py:39]: Can we replace the markdown with `st.subheader`? Would make our code more easily extendable

**Review by SashaWeinstein — APPROVED** [review]
I think this is good to be merged in as is. I have a couple comments but they aren't necessary. Did you decide whether to add the expected borough from bbl/cd columns?

File: src/colp/components/geospatial_check.py (hunk 1 above)
Comment (Oysters1874) [reply src/colp/components/geospatial_check.py:27]: > I think this is good to be merged in as is. I have a couple comments but they aren't necessary. Did you decide whether to add the expected borough from bbl/cd columns?

Yea, we can do that. It is a great suggestion. We need to implement that on the COLP side. I can take a look.

Comment (SashaWeinstein) [reply src/colp/components/geospatial_check.py:30]: ok sick! might be good to link the issue in this convo

**Review by SashaWeinstein — APPROVED** [review]
Looks good to me!

**Pull request merged**